ballerina/soap_utils.bal

```diff
@@ -150,16 +150,20 @@ isolated function createSoap11HttpRequest(xml|mime:Entity[] body, string soapAct
 isolated function createSoap12HttpRequest(xml|mime:Entity[] body, string? soapAction,
                                           map<string|string[]> headers = {}) returns http:Request {
     http:Request req = new;
-    if body is xml {
-        req.setXmlPayload(body);
+    _ = body is xml ? req.setXmlPayload(body) : req.setBodyParts(body);
+    if soapAction is string {
+        map<string> stringMap = {};
+        stringMap["action"] = "\"" + soapAction + "\"";
+        var mediaType = mime:getMediaType(mime:APPLICATION_SOAP_XML);
+        if mediaType is mime:MediaType {
+            mediaType.parameters = stringMap;

```

Comment:
```suggestion
            mediaType.parameters = {"action": string`"${soapAction}"`};

```

Reply:
Fixed in [703e5bb](https://github.com/ballerina-platform/module-ballerina-soap/pull/116/commits/703e5bbc039b4c7ec5d486e287916837506d7f7e)